biogrid/libs/biogrid-simulator/src/lib/biobattery/biobattery-class.ts

```diff
@@ -24,25 +33,18 @@ export class BioBattery implements Battery {
    * @param x Distance from the left edge of the town
    * @param y Distance from the top edge of the town
    */
-  constructor(
-    x: Distance,
-    y: Distance,
-    gridItemName: string,
-    gridItemResistance: number,
-    energyInJoules: Energy = SMALL_BATTERY.DEFAULT_START_ENERGY,
-    maxCapacity: Energy = SMALL_BATTERY.MAX_CAPACITY
-  ) {
-    this.relativePosition = { x, y };
-    if (!this.validateInputs(energyInJoules, maxCapacity)) {
+  constructor(batteryParams: BatteryParams) {
+    this.relativePosition = { x: batteryParams.x, y: batteryParams.y };
+    if (!this.validateInputs(batteryParams.energyInJoules, batteryParams.maxCapacity)) {
       // TODO return a tuple of from validate to with the boolean and unpassed validations
       throw new Error(
-        `Cannot create a battery with values: (${energyInJoules}, ${maxCapacity})`
+        `Cannot create a battery with values: (${batteryParams.energyInJoules}, ${batteryParams.maxCapacity})`
       );
     }
-    this.energyInJoules = energyInJoules;
-    this.maxCapacity = maxCapacity;
-    this.gridItemName = gridItemName;
-    this.gridItemResistance = gridItemResistance;
+    this.energyInJoules = batteryParams.energyInJoules;
+    this.maxCapacity = batteryParams.maxCapacity ? batteryParams.maxCapacity : SMALL_BATTERY.MAX_CAPACITY;
```

Comment:
I believe you can consolidate this to
`this.maxCapacity = batteryParams.maxCapacity || SMALL_BATTERY.MAX_CAPACITY;`

Reply:
To be more consistent with how you handle optional params later in this PR, maybe best to do
```
export class BioBattery implements Battery {
  private readonly maxCapacity: Energy = SMALL_BATTERY.MAX_CAPACITY;
  ...
  constructor(...) {
    ...
    if (batteryParams.maxCapacity) {
      this.maxCapacity = batteryParams.maxCapacity;
    }
  }
}
```